freight/models/app.py

```diff
@@ -82,3 +82,36 @@ def get_default_ref(self, env):
         if not data:
             return DEFAULT_REF
         return data.get('default_ref', DEFAULT_REF)
+
+    def get_current_sha(self, env):
```

Comment:
less code in views, not a big fan of private model methods in views but I respect your point of view.